src/visitors/defuse_analyze_visitor.cpp

```diff
@@ -86,11 +86,14 @@ std::string DUChain::to_string(bool compact) const {
  *  As these are innermost blocks, we have to just check first use
  *  of variable in this block and that's the result of this block.
  */
-DUState DUInstance::sub_block_eval() const {
+DUState DUInstance::sub_block_eval(DUVariableType variable_type = DUVariableType::Global) const {
     DUState result = DUState::NONE;
     for (const auto& chain: children) {
-        const auto& child_state = chain.eval();
-        if (child_state == DUState::U || child_state == DUState::D) {
+        const auto& child_state = chain.eval(variable_type);
+        if ((variable_type == DUVariableType::Global &&
+             (child_state == DUState::U || child_state == DUState::D)) ||
+            (variable_type == DUVariableType::Local &&
+             (child_state == DUState::LU || child_state == DUState::LD))) {
```

Comment:
can we have a `DUState::LU` or `DUState::LD` without being a local variable? Otherwise this can be simplified

Reply:
The discrimination between `DUVariableType::Global` and `DUVariableType::Local` is needed so that we can discard any `LD` or `LU` of the variable we are analyzing if it's not a `local_var` on the Symbol Table. I couldn't think of any better way to do this while being as explanatory as possible. It's certainly a quite complex analysis

Reply:
Just as a note: we discussed different alternatives to avoid introducing new `DUVariableType` (e.g. by using symbol table `current_symtab` directly. That avoids new type but we need some way to know the type). 

We can go ahead with this implementation and revisit this if we see more use cases requiring further refactoring.